test/integration/protocol_integration_test.cc

```diff
@@ -644,6 +648,29 @@ TEST_P(DownstreamProtocolIntegrationTest, MissingHeadersLocalReply) {
   EXPECT_THAT(waitForAccessLog(access_log_name_), HasSubstr("InvalidHeaderFilter_ready\n"));
 }
 
+TEST_P(DownstreamProtocolIntegrationTest, MissingHeadersLocalReplyBytesCount) {
+  useAccessLog("%DOWNSTREAM_WIRE_BYTES_SENT% %DOWNSTREAM_WIRE_BYTES_RECEIVED% "
+               "%DOWNSTREAM_HEADER_BYTES_SENT% %DOWNSTREAM_HEADER_BYTES_RECEIVED%\n");
+  config_helper_.addFilter("{ name: invalid-header-filter, typed_config: { \"@type\": "
+                           "type.googleapis.com/google.protobuf.Empty } }");
+  initialize();
+  codec_client_ = makeHttpConnection(lookupPort("http"));
+
+  // Missing method
+  auto response = codec_client_->makeHeaderOnlyRequest(
+      Http::TestRequestHeaderMapImpl{{":method", "GET"},
+                                     {":path", "/test/long/url"},
+                                     {":scheme", "http"},
+                                     {":authority", "host"},
+                                     {"remove-method", "yes"},
+                                     {"send-reply", "yes"}});
+  ASSERT_TRUE(response->waitForEndStream());
+  EXPECT_TRUE(response->complete());
+  EXPECT_EQ("200", response->headers().getStatusValue());
+  expectDownstreamBytesSentAndReceived(BytesCountExpectation(90, 80, 71, 46),
+                                       BytesCountExpectation(0, 58, 0, 58));
```

Comment:
So this is due to us dispatching the client bytes when we begin the local reply which encodes the response headers. Since we're `dispatching_` `sendPendingFrames()` becomes a no-op e.g. we don't serialize the frames and call `onFrameSend()`. We then continue from `maybeEndEncode->endStream->doEndStream->doDeferredStreamDestroy-> access loggers emitting / filter chain destroy`. 

Later on the frame gets serialized and sent. This is why we're missing those bytes. 

If we're able to have the serialization occur in `sendPendingFrame()`it includes these bytes as that'll call `onFrameSend`. The question is how to do this safely.

Reply:
The way we parse bytes counts from access log require that we make it a separate test. because we only expect the bytes number in the access log, while the origin test expect RESPONSE_CODE_DETAILS in the access log.